tools/inference/openvino_inference.py

```diff
@@ -4,16 +4,145 @@
 file system, and show the visualization results.
 """
 
-# Copyright (C) 2022 Intel Corporation
+# Copyright (C) 2023 Intel Corporation
 # SPDX-License-Identifier: Apache-2.0
 
+from __future__ import annotations
+
 import warnings
 from argparse import ArgumentParser, Namespace
 from pathlib import Path
+from typing import Iterator, Tuple
+
+import cv2
+import numpy as np
+
+try:
+    from openvino.model_api.models import AnomalyDetection, AnomalyResult
+except ImportError:
+    raise ImportError("Ensure that OpenVINO model zoo is installed in your environment.")
+
+
+class Visualizer:
+    @staticmethod
+    def superimpose_pred_boxes(image: np.ndarray, pred_boxes: np.ndarray) -> np.ndarray:
+        """Superimpose predicted boxes on image.
+
+        Args:
+            image (np.ndarray): Input image.
+            pred_boxes (np.ndarray): Predicted boxes.
+
+        Returns:
+            np.ndarray: Image with superimposed boxes.
+        """
+        for box in pred_boxes:
+            if len(box) == 4:
+                image = cv2.rectangle(
+                    image,
+                    (box[0], box[1]),
+                    (box[2], box[3]),
+                    (0, 255, 0),
+                    2,
+                )
+        return image
+
+    @staticmethod
+    def superimpose_anomaly_map(anomaly_map: np.ndarray, image: np.ndarray) -> np.ndarray:
+        """Superimpose anomaly map on the input image.
+
+        Args:
+            anomaly_map (np.ndarray): Anomaly map.
+            image (np.ndarray): Input image.
+
+        Returns:
+            np.ndarray: Superimposed image.
+        """
+        # convert to color map
+        anomaly_map *= 255
```

Comment:
should we have check here for the pixel range?

Reply:
Good idea